include/quda.h

```diff
@@ -1013,6 +1013,16 @@ extern "C" {
    */
   void updateMultigridQuda(void *mg_instance, QudaMultigridParam *param);
 
+  /**
+   * @brief Performs a thin update to the multigrid preconditioner for the new gauge / clover field.
+   * This update only recreates the Dirac classes and does not rebuild the entire coarse setup.
+   * Assumes gauge parameters haven't changed.
+   * @param mg_instance Pointer to instance of multigrid_solver
+   * @param param Contains all metadata regarding host and device
+   * storage and solver parameters
+   */
+  void thinUpdateMultigridQuda(void *mg_instance, QudaMultigridParam *param);
+
```

Comment:
I'm not sure I understand why we need a new interface for this functionality.  Why can't this be parameterized and deployed in the already existing `updateMultigridQuda` interface function?  This approach is much more intrusive and would require more extensive changes in case, for example was wanted to deploy a "thin" update in other applications.

Reply:
Two part reply:

1. The core of the issue here is identical to what we discussed on Slack for #1018, extending an existing interface with new functionality, where @mathiaswagner introduced the tentative solution of adding `qudaGaugeForcePhased` alongside `qudaGaugeForce`. The issues we discussed were:
* Not wanting to modify the signature of `qudaGaugeForce` to avoid breaking existing interfaces
* C not supporting default arguments, so we couldn't just add an int/enum/etc corresponding to phased vs unphased (in that context), or thin vs full here.
* Not wanting to modify a data structure passed in to avoid breaking existing interfaces (the issue there more being that structures aren't guaranteed to be zero-initialized, so you risk ending up with undefined behavior).

As I said, that solution was tentative, so there's room for further discussion that would apply in both places. If we're willing to break long-standing interfaces (`qudaGaugeForce` in #1018, `updateMultigridQuda` here), then things become a lot easier, but I don't like that approach.

2. In the sense that this is MG-specific, I'm certainly open to modifying this signature to be more future-proof.

Reply:
I still don't understand this.  Why can't any objectives be implemented through parameters in `QudaMultigridParam`?  The issue is different from #1018 since this is a QUDA interface, not a MILC-QUDA interface.

Reply:
We can simply ensure that the default behaviour matches the present `updateMultigridQuda`.  Perhaps I am missing some subtlety? (or not so subtlety 😜 )

Reply:
Ah, fair point. I can add it to `QudaMultigridParam`.

Reply:
Schematically done, though it may not compile yet ;)